app/bundles/CategoryBundle/Controller/CategoryController.php

```diff
@@ -310,7 +310,22 @@ public function editAction(Request $request, $bundle, $objectId, $ignorePost = f
         } elseif (!$this->security->isGranted($model->getPermissionBase($bundle).':view')) {
             return $this->modalAccessDenied();
         } elseif ($model->isLocked($entity)) {
-            return $this->modalAccessDenied();
+            $viewParams = [
+                'page'   => $session->get('mautic.category.page', 1),
+                'bundle' => $bundle,
+            ];
+            $postActionVars = [
+                'returnUrl'       => $this->generateUrl('mautic_category_index', $viewParams),
+                'viewParameters'  => $viewParams,
+                'contentTemplate' => 'MauticCategoryBundle:Category:index',
```

Comment:
Since we moved away from `templating` this line should be like this

```suggestion
                'contentTemplate' => '@MauticCategory/Category/index.html.twig',
```